scripts/convert_llama_to_dcp.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+import json
+from pathlib import Path
+
+import torch
+import torch.distributed.checkpoint as DCP
+
+@torch.inference_mode()
+def convert_llama_weights(input_dir, output_dir):
+    with open(args.input_dir / "params.json", "r") as f:
+        params = json.load(f)
+    n_layers = params["n_layers"]
+    n_heads = params["n_heads"]
+    dim = params["dim"]
+    dims_per_head = dim // n_heads
+
+    checkpoint_list = sorted([file for file in input_dir.rglob("*.pth")])
+    print(f"Loading from these files: {checkpoint_list}.")
+    shards = [torch.load(ckpt, map_location="cpu", weights_only=True) for ckpt in checkpoint_list]
+
+    n_heads_per_shard = n_heads // len(shards)
+    num_key_value_heads = params["n_kv_heads"]
+    n_kv_heads_per_shard = num_key_value_heads // len(shards)
+    key_value_dim = dims_per_head * num_key_value_heads
+
+    if len(shards) == 1:
+        state_dict = shards[0]
+    else: # sharded
+        state_dict = {}
+        for layer in range(n_layers):
+            state_dict[f"layers.{layer}.attention_norm.weight"] = shards[0][f"layers.{layer}.attention_norm.weight"].clone() # replicated
+            state_dict[f"layers.{layer}.ffn_norm.weight"] = shards[0][f"layers.{layer}.ffn_norm.weight"].clone() # replicated
```

Comment:
Is it necessary to do `.clone()` here? It seems for `norm.weight` below there is no cloning.